src/Libraries/Microsoft.Extensions.AI.Abstractions/Embeddings/EmbeddingGeneratorMetadata.cs

```diff
@@ -48,4 +68,43 @@ public EmbeddingGeneratorMetadata(string? providerName = null, Uri? providerUri
     /// An individual request may override this value via <see cref="EmbeddingGenerationOptions.Dimensions"/>.
     /// </remarks>
     public int? Dimensions { get; }
+
+    /// <summary>
+    /// Gets a read-only list of media types (also known as MIME types) the generator has declared to support as inputs.
+    /// </summary>
+    /// <remarks>
+    /// If <see langword="null"/>, the generator has not made any declarations about supported types.
+    /// If non-<see langword="null"/>, the generator supports only the listed types, and providing data
+    /// that is of another type may lead to exceptions or unexpected results.
+    /// </remarks>
+    public ISet<string>? SupportedInputMediaTypes { get; }
```

Comment:
It's a bit odd to use media types to define the set of supported input types. Media types, as a concept, only apply to serialized data and a `TInput` isn't necessarily any kind of serialized data.

Reply:
I guess more broadly I'm not clear on how this information would get used. In this PR, I don't see examples of it getting used. Could you give an example of how application code would interact with this?